### PR TITLE
Fix memory orders and namespacestd

### PR DESCRIPTION
**Pull request by DylanSchneider**
https://logrhythm.atlassian.net/browse/NM-1297

### CODE REVIEW

**Comment by KjellKod** [comment]
Are you sure you want to keep the loading as sequential load now that you use std::memory_order_acquire  for the store? 

Ref: https://en.cppreference.com/w/cpp/atomic/memory_order

Hope you are doing well in Boulder/LogRhythm :)

**Comment by DylanSchneider** [comment]
> Are you sure you want to keep the loading as sequential load now that you use std::memory_order_acquire for the store?
> 
> Ref: https://en.cppreference.com/w/cpp/atomic/memory_order
> 
> Hope you are doing well in Boulder/LogRhythm :)

What a gem you are, Kjell,
I missed the `load` call in StopWatch, so thank you for pointing that out. We decided to load/store like this based on Herb Sutter's video we watched with you. So you are correct, and it should be fixed in my PR now. Thanks again!

File: src/AlarmClock.h [hunk 1]
```diff
@@ -77,7 +76,7 @@ template<typename Duration> class AlarmClock {
       // the two atomics. Therefore stopwatch is needed to ensure the alarm
       // does not over sleep. 
       while (sw.ElapsedUs() < timeUsTillExpire) {
-         this_thread::sleep_for(microseconds(25));
+         std::this_thread::sleep_for(microseconds(25));
          if (mReset || mExit) {
             return false;
```
Comment (KjellKod) [inline src/AlarmClock.h]: @DylanSchneider
Please do a search for all atomic members and verify their usage

Example: (pseudocode)
```
if (mReset.load(aquire) || mExit.load(aquire)
```

File: src/AlarmClock.h [hunk 2]
```diff
@@ -62,12 +61,12 @@ template<typename Duration> class AlarmClock {
    void AlarmClockInterruptableThread() {
       while(!mExit) {
```
Comment (john-gress) [inline src/AlarmClock.h]: `mExit.load(std::memory_order_acquire)`

File: src/AlarmClock.h [hunk 3]
```diff
@@ -62,12 +61,12 @@ template<typename Duration> class AlarmClock {
...
          if(mAlarmExpiredFunction(kSleepTimeUsCount)) {
-            mExpired.store(true);
+            mExpired.store(true, std::memory_order_release);
             while (!mReset && !mExit) {
```
Comment (john-gress) [inline src/AlarmClock.h]: `while (!mReset.load(std::memory_order_acquire) && !mExit.load(std::memory_order_acquire))`

File: src/AlarmClock.h [hunk 4]
```diff
@@ -77,7 +76,7 @@ template<typename Duration> class AlarmClock {
       // the two atomics. Therefore stopwatch is needed to ensure the alarm
       // does not over sleep. 
       while (sw.ElapsedUs() < timeUsTillExpire) {
-         this_thread::sleep_for(microseconds(25));
+         std::this_thread::sleep_for(microseconds(25));
          if (mReset || mExit) {
```
Comment (john-gress) [inline src/AlarmClock.h]: `if (mReset.load(std::memory_order_acquire) || mExit.load(std::memory_order_acquire))`

**Comment by john-gress** [comment]
@KjellKod, Thanks for reviewing these changes. You're awesome!

**Pull request merged**